library/std/src/sys_common/backtrace.rs

```diff
@@ -85,10 +87,27 @@ unsafe fn _print_fmt(fmt: &mut fmt::Formatter<'_>, print_fmt: PrintFmt) -> fmt::
                         start = true;
                         return;
                     }
+                    if !start {
+                        omitted_count += 1;
+                    }
                 }
             }
 
             if start {
+                if omitted_count > 0 {
+                    assert!(print_fmt == PrintFmt::Short);
+                    // only print the message between the middle of frames
+                    if !first_omit {
+                        let omit_message = format!(
+                            "      [... omitted {} frame{} ...]\n",
+                            omitted_count,
+                            if omitted_count > 1 { "s" } else { "" }
+                        );
+                        let _ = bt_fmt.message(&omit_message);
```

Comment:
This makes me think it might be better if `BacktraceFmt` had a method for exposing its `Formatter`, such that you can directly `write!()` into it without allocating a string. (Or, alternatively, maybe `BacktraceFmt` could implement `fmt::Write` directly, forwarding to its `Formatter`.)

Reply:
fixed :)